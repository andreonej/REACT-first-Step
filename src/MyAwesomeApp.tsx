import { type CSSProperties } from 'react';
import styles from './MyAwesomeApp.module.css';

// Datos del usuario organizados en un objeto
const userProfile = {
  firstName: 'Jose',
  lastName: 'Andeone',
  favoriteGames: ['Elden Ring', 'Super Smash Bros', 'Metal Gear Solid'],
  isActive: true,
  address: {
    zipCode: 'ABC-123',
    country: 'Canad치',
    city: 'Toronto'
  },
  joinDate: new Date('2023-01-15')
} as const;

const cardStyles: CSSProperties = {
  background: 'linear-gradient(135deg, #667eea 0%, #764ba2 100%)',
  borderRadius: '16px',
  padding: '2rem',
  color: 'white',
  boxShadow: '0 8px 32px rgba(0, 0, 0, 0.1)'
};

export const MyAwesomeApp = () => {
  const formatDate = (date: Date) => {
    return date.toLocaleDateString('es-ES', {
      year: 'numeric',
      month: 'long',
      day: 'numeric'
    });
  };

  return (
    <div className={styles.container}>
      <div className={styles.profileCard} style={cardStyles}>
        <header className={styles.header}>
          <div className={styles.avatar}>
            {userProfile.firstName.charAt(0)}{userProfile.lastName.charAt(0)}
          </div>
          <div className={styles.userInfo}>
            <h1 className={styles.fullName} data-testid="first-name-title">
              {userProfile.firstName} {userProfile.lastName}
            </h1>
            <div className={styles.status}>
              <span className={`${styles.statusIndicator} ${userProfile.isActive ? styles.active : styles.inactive}`}></span>
              {userProfile.isActive ? 'En l칤nea' : 'Desconectado'}
            </div>
          </div>
        </header>

        <section className={styles.section}>
          <h3 className={styles.sectionTitle}>游꿡 Juegos Favoritos</h3>
          <ul className={styles.gamesList}>
            {userProfile.favoriteGames.map((game, index) => (
              <li key={index} className={styles.gameItem}>
                {game}
              </li>
            ))}
          </ul>
        </section>

        <section className={styles.section}>
          <h3 className={styles.sectionTitle}>游늸 Informaci칩n de Ubicaci칩n</h3>
          <div className={styles.addressInfo}>
            <p><strong>Pa칤s:</strong> {userProfile.address.country}</p>
            <p><strong>Ciudad:</strong> {userProfile.address.city}</p>
            <p><strong>C칩digo Postal:</strong> {userProfile.address.zipCode}</p>
          </div>
        </section>

        <footer className={styles.footer}>
          <p className={styles.joinDate}>
            Miembro desde: {formatDate(userProfile.joinDate)}
          </p>
        </footer>
      </div>
    </div>
  );
};