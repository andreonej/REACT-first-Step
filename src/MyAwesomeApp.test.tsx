import { describe, expect, test } from 'vitest';
import { render, screen } from '@testing-library/react';

import { MyAwesomeApp } from './MyAwesomeApp';

describe('MyAwesomeApp', () => {
  test('should render firstName and lastName', () => {
    const { container } = render(<MyAwesomeApp />);
    //screen.debug();
    //console.log(container.innerHTML);

    const h1 = container.querySelector('h1');
    const h3 = container.querySelector('h3');

    expect(h1?.innerHTML).toContain('Jose');
    expect(h3?.innerHTML).toContain('ðŸŽ® Juegos Favoritos');
  });

   test('should render firstName and lastName - screen', () => {
     render(<MyAwesomeApp />);
     //screen.debug();
    // console.log(container.innerHTML);
    // const h1 = screen.getByRole('heading', {
    //   level: 1,
    // });
    const h1 = screen.getByTestId('first-name-title'); // No es recomenda usar ID-Test
     expect(h1.innerHTML).toContain('Jose');
   });

  test('should match snapshot', () => {
    const { container } = render(<MyAwesomeApp />);
    expect(container).toMatchSnapshot(); // Toma una foto de la app, se busca para validar que no cambia la apariencia.
  });

   test('should match snapshot', () => {
      render(<MyAwesomeApp />);
      expect(screen.getByTestId('first-name-title')).toMatchSnapshot();
   });
});
