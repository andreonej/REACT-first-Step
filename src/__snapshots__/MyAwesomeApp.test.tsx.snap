// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MyAwesomeApp > should match snapshot 1`] = `
<div>
  <div
    class="_container_b8db5f"
  >
    <div
      class="_profileCard_b8db5f"
      style="background: linear-gradient(135deg, #667eea 0%, #764ba2 100%); border-radius: 16px; padding: 2rem; color: white; box-shadow: 0 8px 32px rgba(0, 0, 0, 0.1);"
    >
      <header
        class="_header_b8db5f"
      >
        <div
          class="_avatar_b8db5f"
        >
          J
          A
        </div>
        <div
          class="_userInfo_b8db5f"
        >
          <h1
            class="_fullName_b8db5f"
            data-testid="first-name-title"
          >
            Jose
             
            Andeone
          </h1>
          <div
            class="_status_b8db5f"
          >
            <span
              class="_statusIndicator_b8db5f _active_b8db5f"
            />
            En l칤nea
          </div>
        </div>
      </header>
      <section
        class="_section_b8db5f"
      >
        <h3
          class="_sectionTitle_b8db5f"
        >
          游꿡 Juegos Favoritos
        </h3>
        <ul
          class="_gamesList_b8db5f"
        >
          <li
            class="_gameItem_b8db5f"
          >
            Elden Ring
          </li>
          <li
            class="_gameItem_b8db5f"
          >
            Super Smash Bros
          </li>
          <li
            class="_gameItem_b8db5f"
          >
            Metal Gear Solid
          </li>
        </ul>
      </section>
      <section
        class="_section_b8db5f"
      >
        <h3
          class="_sectionTitle_b8db5f"
        >
          游늸 Informaci칩n de Ubicaci칩n
        </h3>
        <div
          class="_addressInfo_b8db5f"
        >
          <p>
            <strong>
              Pa칤s:
            </strong>
             
            Canad치
          </p>
          <p>
            <strong>
              Ciudad:
            </strong>
             
            Toronto
          </p>
          <p>
            <strong>
              C칩digo Postal:
            </strong>
             
            ABC-123
          </p>
        </div>
      </section>
      <footer
        class="_footer_b8db5f"
      >
        <p
          class="_joinDate_b8db5f"
        >
          Miembro desde: 
          14 de enero de 2023
        </p>
      </footer>
    </div>
  </div>
</div>
`;

exports[`MyAwesomeApp > should match snapshot 2`] = `
<h1
  class="_fullName_b8db5f"
  data-testid="first-name-title"
>
  Jose
   
  Andeone
</h1>
`;
